src/sidebar/build-thread.js

```diff
@@ -2,107 +2,106 @@
+ * @prop {Annotation} [annotation] - This thread's annotation. Undefined in cases
+ *       when an annotation _should_ exist—it's implied by a reference from
+ *       another annotation—but is not present in our collection of annotations
+ * @prop {string} [parent] - The id of this thread's parent. Top-level threads
+ *       do not have parents
+ * @prop {boolean} visible - Whether this thread should be visible when rendered.
+ *       true when the thread's annotation matches current annotation filters.
+ * @prop {boolean} collapsed - Whether the replies in this thread should be
+ *       rendered as collapsed (when true) or expanded (when false)
  * @prop {Thread[]} children
- * @prop {number} totalChildren
- * @prop {'dim'|'highlight'|undefined} highlightState
+ * @prop {number} totalChildren - Computed count of this thread's immediate
+ *       children. This count includes visually-hidden threads.
+ * @prop {number} replyCount - Computed count of all replies to a thread
+ * @prop {'dim'|'highlight'} [highlightState] - In cases where there are one
+ *       or more threads currently designated as "highlighted", we track
+ *       the highlight state for each thread. undefined when there are no
+ *       highlighted threads.
  */
 
 /**
- * Default state for new threads, before applying filters etc.
- *
- * @type {Thread}
+ * Default state for new threads
  */
 const DEFAULT_THREAD_STATE = {
-  /**
-   * The ID of this thread. This will be the same as the annotation ID for
-   * created annotations or the `$tag` property for new annotations.
-   */
   id: '__default__',
-  /**
-   * The Annotation which is displayed by this thread.
-   *
-   * This may be null if the existence of an annotation is implied by the
-   * `references` field in an annotation but the referenced parent annotation
-   * does not exist.
-   */
-  annotation: undefined,
-  /** The parent thread ID */
-  parent: undefined,
-  /** True if this thread is collapsed, hiding replies to this annotation. */
   collapsed: false,
-  /** True if this annotation matches the current filters. */
   visible: true,
-  /** Replies to this annotation. */
-  children: [],
-  /**
-   * The total number of children of this annotation,
-   * including any which have been hidden by filters.
-   */
+  replyCount: 0,
   totalChildren: 0,
-  /**
-   * The highlight state of this annotation:
-   *  undefined - Do not (de-)emphasize this annotation
-   *  'dim' - De-emphasize this annotation
-   *  'highlight' - Emphasize this annotation
-   */
-  highlightState: undefined,
 };
 
 /**
  * Returns a persistent identifier for an Annotation.
  * If the Annotation has been created on the server, it will have
- * an ID assigned, otherwise we fall back to the local-only '$tag'
+ * an id assigned, otherwise we fall back to the local-only '$tag'
  * property.
+ *
+ * @param {Annotation} annotation
+ * @return {string}
  */
-function id(annotation) {
+function annotationId(annotation) {
```

Comment:
Right now this module has no dependencies, which is nice, but I feel like this metadata utility function would ideally be in a shared place.